### ISSUE TITLE
Kubernetes role remapping does not work in root/leaf clusters

### ISSUE THREAD

**Issue by ungureanuvladvictor** [gravitational/teleport#4270]
### Description

My setup is the following:
- root cluster hosted in AWS spinned up from the example that you folks have in the repo (via Terraform)
- leaf cluster running in a separate AWS account which has a full blown k8s installation
- in the root cluster I login and get the role of admin. The admin role does not include any kubernetes users or groups in its definition
- in the leaf cluster I am remapping all the roles that come from the root to the local admin role which includes `system:masters` in its `kubernetes_groups` field

**What happened**:
After I do a `tsh login --format=kubernetes $LEAFCLUSTER` I get a kubeconfig setup correctly for my leaf. If I do `kubectl get pods --all-namespaces` I get back a 403 `Error from server (Forbidden): unknown (get pods)`. Looking at the root cluster proxy logs I can see the followqing:
```
Aug 28 22:49:01 ip-172-31-2-246.ec2.internal /usr/bin/teleport[13654]: WARN [PROXY:KUB] this user cannot request kubernetes access, has no assigned groups or users proxy/forwarder.go:313
```

If I add the `system:masters` group to the admin role in the root cluster then the kubectl cmd works as expected.

**What you expected to happen**:
I expected the role remapping to work even if the `kubernetes_groups` field was not specified in the root cluster admin role definition. 

**How to reproduce it (as minimally and precisely as possible)**:
- spin up a root/leaf cluster setup
- leaf should run k8s
- create a dummy role in the root which does not have `kubernetes_groups` field specified at all
- create a role mapping in the leaf assigning all roles from the root to admin on the leaf
- login via tsh specifying the format kubernetes and connecting to the leaf cluster
- run kubectl cmds and experience the 403 error

### Environment
- Teleport version (use `teleport version`): 4.3.5
- Tsh version (use `tsh version`): 4.3.5
- OS (e.g. from `/etc/os-release`): leaf on `18.04.5 LTS (Bionic Beaver)`, root on `Amazon Linux 2`

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS (both leaf and root)

**Browser environment**

- Browser Version (for UI-related issues):
- Install tools:
- Others:

**Relevant Debug Logs If Applicable**
Don't have more debug info except the error I got from the proxy in the root cluster which I added above.

**Comment by webvictim** [comment]
Does this work as expected if you add `{{internal.kubernetes_groups}}` to the `kubernetes_groups` section of the role on your root cluster?

You’ll need to log out with `tsh logout` and then `tsh login` again after making the change.

**Comment by webvictim** [comment]
(this is potentially related to https://github.com/gravitational/teleport/issues/3376 as well)

**Comment by ungureanuvladvictor** [comment]
@webvictim I tried your suggestion but it does not work. In the root cluster proxy logs I still see:
```
Sep 02 03:37:43 ip-172-31-2-246.ec2.internal /usr/bin/teleport[13654]: WARN [PROXY:KUB] this user cannot request kubernetes access, has no assigned groups or users proxy/forwarder.go:313
```

**Comment by awly** [comment]
@ungureanuvladvictor this is a bit hacky, but can you add any dummy kubernetes users or groups to your root cluster roles?
For example:
```
kind: role
version: v3
metadata:
  name: admin
spec:
  allow:
    kubernetes_groups: ["dummy_group_do_not_grant_any_permissions"]
```
The group will be overwritten by the leaf cluster role, this is just to bypass the flawed logic in the root proxy.

I'll fix this behavior in 5.0, since I believe the above workaround will get you unblocked.
Let me know if it doesn't!

**Comment by ungureanuvladvictor** [comment]
@awly yep, this works

**Comment by awly** [comment]
This appears to be the same issue as https://github.com/gravitational/teleport/issues/3376 and will be fixed in 5.2.
I'll close this issue in favor of the older one.
